### PR TITLE
Add flag  that allows to automatically confirm bookings created by Bo…

### PR DESCRIPTION
**Pull request by cekk**
…oking Managers

### CODE REVIEW

File: src/redturtle/prenotazioni/adapters/booker.py [hunk 1]
```diff
@@ -71,6 +71,15 @@ def _validate_user_limit(self, data: dict):
         ) >= (self.context.max_bookings_allowed):
             raise BookingsLimitExceded(self.context, booking_type=data["booking_type"])
 
+    @property
+    def is_gestore(self):
```
Comment (mamico) [inline src/redturtle/prenotazioni/adapters/booker.py:75]:
```suggestion
    def is_mamanger(self):
```

File: src/redturtle/prenotazioni/adapters/booker.py [hunk 2]
```diff
@@ -251,6 +260,13 @@ def book(self, data, force_gate=None, duration=-1):
         if not obj:
             msg = _("Sorry, this slot is not available anymore.")
             raise BookerException(api.portal.translate(msg))
+
+        if self.is_gestore:
```
Comment (mamico) [inline src/redturtle/prenotazioni/adapters/booker.py:264]:
```suggestion
        if self.is_manager:
```
```suggestion
        if self.is_gestore:
```

File: src/redturtle/prenotazioni/restapi/services/booking/add.py [hunk 3]
```diff
@@ -21,6 +21,12 @@ class AddBooking(BookingSchema):
     Add a new booking
     """
 
+    @property
+    def is_gestore(self):
```
Comment (mamico) [inline src/redturtle/prenotazioni/restapi/services/booking/add.py:25]:
```suggestion
    def is_manager(self):
```

File: src/redturtle/prenotazioni/restapi/services/booking/add.py [hunk 4]
```diff
@@ -65,6 +66,10 @@ def validate(self):
         data = json_body(self.request)
         data_fields = {field["name"]: field["value"] for field in data["fields"]}
 
+        if data.get("force", False) and not self.is_gestore:
```
Comment (mamico) [inline src/redturtle/prenotazioni/restapi/services/booking/add.py:69]:
```suggestion
        if data.get("force", False) and not self.is_manager:
```

**Review by mamico — APPROVED** [review]

File: src/redturtle/prenotazioni/restapi/services/booking/add.py [hunk 5]
```diff
@@ -36,22 +42,17 @@ def reply(self):
...
+                obj = booker.book(
+                    data=book_data,
+                    force_gate=gate,
+                    duration=duration,
+                )
             else:
                 obj = booker.book(data=book_data)
         except BookerException as e:
-            raise BadRequest(str(e))
+            raise BadRequest(e) from e
```
Comment (mamico) [inline src/redturtle/prenotazioni/restapi/services/booking/add.py:55]: ?

Comment (cekk) [reply src/redturtle/prenotazioni/restapi/services/booking/add.py:55]: i had this suggestion in vscode: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/raise-missing-from.html

**Pull request merged**